neuralpy/layers/convolutional/convtranspose1d.py

```diff
@@ -0,0 +1,238 @@
+        self.__groups = groups
+        self.__bias = bias
+        self.__dilation = dilation
+        self.__name = name
+
+    def __get_layer_details(self):
+        # Return tuple structure
+        # Getting the kernel values
+        kernel1 = 0
+        if isinstance(self.__kernel_size, int):
+            kernel1 = self.__kernel_size
+        else:
+            kernel1, = self.__kernel_size
+
+        # Getting the padding values
+        padding1 = 0
+        if isinstance(self.__padding, int):
+            padding1 = self.__padding
+        else:
+            padding1, = self.__padding
+
+        # Getting the stride values
+        stride_1 = 0
+        if isinstance(self.__stride, int):
+            stride_1 = self.__stride
+        else:
+            stride_1, = self.__stride
+
+        # Getting the dilation values
+        dilation1 = 0
+        if isinstance(self.__dilation, int):
+            dilation1 = self.__dilation
+        else:
+            dilation1, = self.__dilation
+
+        # Getting the out_padding values
+        out_padding1 = 0
+        if isinstance(self.__out_padding, int):
+            out_padding1 = self.__out_padding
+        else:
+            out_padding1, = self.__out_padding
+
+        # Calculating the width and height of the convtranspose output
+        dim1 = (
+            (self.__input_shape[-1] - 1)
+            * (stride_1 - 2)
+            * padding1
+            + dilation1
+            * (kernel1 - 1)
+            + (out_padding1 + 1)
+        )
+
+        return(
+            self.__input_shape[0], self.__in_channels * dim1,
+            (self.__in_channels, dim1)
+        )
+
+    def get_input_dim(self, prev_input_dim, prev_layer_type):
+        """
+        This method calculates the input shape for layer based on previous output
+        layer.
+
+        This method is used by the NeuralPy Models, for building the models.
+        No need to call this method for using NeuralPy.
+        """
+        # Checking if n_inputs is there or not, not overwriting the n_input
+        # field
+        if not self.__input_shape:
+            layer_type = prev_layer_type.lower()
+
+            # based on the prev layer type, predicting the __input_shape
+            # to support more layers, we need to add some more statements
+            if layer_type == 'convtranspose1d':
+                self.__input_shape = prev_input_dim[-1]
+            else:
+                raise ValueError(
+                    "Unsupported previous layer, please provide your own input shape\
+                        for the layer"
+                    )
+
+    def get_layer(self):
+        """
+        This method returns the details as dict of the layer.
+
+        This method is used by the NeuralPy Models, for building the models.
+        No need to call this method for using NeuralPy.
+        """
+        # Returning all the details of the layer
+        return self._get_layer_details(
```

Comment:
The return statement seems incorrect, please check

Reply:
@imdeepmind I can't figure out what I'm missing here. Can you please tell me what I'm missing here?

Reply:
Sorry, it seems fine. By mistake the method `__get_layer_details` and `_get_layer_details` are same, so got confused.